sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/KeyAsyncClient.java

```diff
@@ -1228,8 +1228,14 @@ public String getVaultUrl() {
             return service.getKeyVersions(vaultUrl, name, DEFAULT_MAX_PAGE_RESULTS, apiVersion, ACCEPT_LANGUAGE,
                 CONTENT_TYPE_HEADER_VALUE, context.addData(AZ_TRACING_NAMESPACE_KEY, KEYVAULT_TRACING_NAMESPACE_VALUE))
                 .doOnRequest(ignored -> logger.info("Listing key versions - {}", name))
-                .doOnSuccess(response -> logger.info("Listed key versions - {}", name))
-                .doOnError(error -> logger.warning(String.format("Failed to list key versions - {}", name), error));
+                .doOnSuccess(response -> {
+                    if (response.getValue().size() == 0) {
+                        throw new ResourceNotFoundException(String.format("Key with name: %s not found", name), null);
```

Comment:
Resource Not Found exception is used to indicate 404 response from the service.
if service doesn't throw 404, we should not alter it here as that will result in SDK overriding service behavior and will need to do it manually everywhere in an endless cycle. 
rather change the javadocs to reflect the actual behavior and follow up with the service if this behaviour feels inconsistent or is a bug.

Reply:
Thanks @g2vinay, I will talk to the service team about this then.

Reply:
Before following up with the service, we should check the service response code.
If they're already sending 404, then we need to properly map that in KeyService.class.

Reply:
The service does not respond with a 404, its response instead has a status code 200 and a body like this:

`{
    "value": [],
    "nextLink": null
}`

Which has the same structure as any response where the given key name was found.

Reply:
The stack trace posted on the linked issue above: 
https://github.com/Azure/azure-sdk-for-java/issues/11033

The issue says 404 is sent by the service.

Reply:
In this [this comment](https://github.com/Azure/azure-sdk-for-java/issues/11033#issuecomment-628369884) I outlined how I could not get a `404` unless I send an empty or null key name, anything else returns `200`, so unless there was a bug with the service at the time or they changed what the behavior is, this is not reproducible with a non-empty key name.